File: WordToDigit/cpp/WordToDigit.cpp
// Word to Digit solution in C++ for CodeEval.com by Steven A. Dunn

#include <fstream>
#include <iostream>
#include <iterator>
#include <string>
#include <vector>

using namespace std;

int createDigit(string);

int main (int argc, char* argv[])
{
	ifstream inputFile(argv[1]);
	string line;
	
	if (inputFile)
	{
		while (getline(inputFile, line))
		{
			vector<string> words;
			size_t semicolonPos = line.find(";");
			
			while (semicolonPos != string::npos)
			{				
				words.push_back(line.substr(0, semicolonPos));
				line = line.substr(semicolonPos + 1);
				semicolonPos = line.find(";");
			}
			words.push_back(line);
			
			for (int i = 0; i < words.size(); i++) 
			{
				cout << createDigit(words[i]);
			}
			cout << endl;
		}
	}
}

int createDigit(string s)
{
	int result = 0;
	if (s == "one") { result = 1; }
	if (s == "two") { result = 2; }
	if (s == "three") { result = 3; }
	if (s == "four") { result = 4; }
	if (s == "five") { result = 5; }
	if (s == "six") { result = 6; }
	if (s == "seven") { result = 7; }
	if (s == "eight") { result = 8; }
	if (s == "nine") { result = 9; }
	return result;
}

File: WordToDigit/c#/WordToDigit.cs
// word to digit soln in c# for code eval by steven a dunn

using System;
using System.IO;

class WordToDigit {
  public static void Main(string[] args) {
    using (StreamReader reader = new StreamReader(args[0])) {
      while (!reader.EndOfStream) {
        string line = reader.ReadLine();
        if (null == line)
          continue;

        string[] inputs = line.Split(';');
        foreach (string s in inputs)
          Console.Write(convert(s));
        Console.WriteLine();
      }
    }
  }

  static int convert(string word) {
    if (word == "one") return 1;
    else if (word == "two") return 2;
    else if (word == "three") return 3;
    else if (word == "four") return 4;
    else if (word == "five") return 5;
    else if (word == "six") return 6;
    else if (word == "seven") return 7;
    else if (word == "eight") return 8;
    else if (word == "nine") return 9;
    else if (word == "zero") return 0;
    else return -1;
  }
}